CITATION.cff

```diff
@@ -0,0 +1,32 @@
+# This CITATION.cff file was generated with cffinit.
+# Visit https://bit.ly/cffinit to generate yours today!
+
```

Comment:
I think ultimately this code is likely to be published somewhere like JOSS (as discussed). If this goes ahead, then we will need to set up archiving via Zenodo (or equivalent) #179. I think we should keep the top bit of the file about the code itself, and the `preferred-citation` (which is what shows in the GUI) about the paper. See https://github.com/astro-informatics/sleplet/blob/main/CITATION.cff.